src/FinSearch.php

```diff
@@ -33,6 +40,34 @@ public function build(ContainerBuilder $container): void
         parent::build($container);
     }
 
+    public function install(InstallContext $installContext): void
+    {
+        parent::install($installContext);
+
+        $currentShopwareVersion = $installContext->getCurrentShopwareVersion();
+        $composerJsonContents = file_get_contents(__DIR__ . '/../composer.json');
+        $parsed = json_decode($composerJsonContents, true);
+        // Do nothing if there is no require property in the json as we probably are in development
+        if (!$parsed) {
+            return;
+        }
+
+        $requiredPackages = $parsed['require'];
+        if (isset($requiredPackages['shopware/core'])) {
+            $compatibleVersions = explode('||', $requiredPackages['shopware/core']);
+            $lowestSupported = ltrim(current($compatibleVersions), '^');
+            $highestSupported = ltrim(end($compatibleVersions), '^');
+            // If the current shopware version does not come in between our lowest and highest supported version,
```

Comment:
This comment is probably not necessary after refactoring. Or as [Uncle Bob](https://twitter.com/unclebobmartin/status/870311898545258497) would say:

> A comment is a failure to express yourself in code.  If you fail, then write a comment; but try not to fail.

Reply:
Very wise Uncle Bob! 👨🏼‍🎓